oracle/types/oracle.go

```diff
@@ -24,113 +23,114 @@ type (
 	// This is responsible for creating the API query handler for the price provider.
 	PriceAPIQueryHandlerFactory = func(
 		logger *zap.Logger,
 		cfg config.ProviderConfig,
 		apiMetrics apimetrics.APIMetrics,
-		pMarketMap ProviderMarketMap,
 	) (PriceAPIQueryHandler, error)
 
 	// PriceAPIFetcher is a type alias for the price API fetcher. This is responsible
 	// for fetching the prices from the price provider.
-	PriceAPIFetcher = apihandlers.APIFetcher[mmtypes.Ticker, *big.Int]
+	PriceAPIFetcher = apihandlers.APIFetcher[ProviderTicker, *big.Int]
 
 	// PriceAPIDataHandler is a type alias for the price API data handler. This
 	// is responsible for parsing http responses and returning the resolved
 	// and unresolved prices.
-	PriceAPIDataHandler = apihandlers.APIDataHandler[mmtypes.Ticker, *big.Int]
+	PriceAPIDataHandler = apihandlers.APIDataHandler[ProviderTicker, *big.Int]
 
 	// PriceAPIQueryHandler is a type alias for the price API query handler. This
 	// is responsible for building the API query for the price provider and
 	// returning the resolved and unresolved prices.
-	PriceAPIQueryHandler = apihandlers.APIQueryHandler[mmtypes.Ticker, *big.Int]
+	PriceAPIQueryHandler = apihandlers.APIQueryHandler[ProviderTicker, *big.Int]
 
 	// PriceWebSocketQueryHandlerFactory is a type alias for the price web socket query handler factory.
 	// This is responsible for creating the web socket query handler for the price provider.
 	PriceWebSocketQueryHandlerFactory = func(
 		logger *zap.Logger,
 		cfg config.ProviderConfig,
 		wsMetrics wsmetrics.WebSocketMetrics,
-		pMarketMap ProviderMarketMap,
 	) (PriceWebSocketQueryHandler, error)
 
 	// PriceWebSocketDataHandler is a type alias for the price web socket data handler.
 	// This is responsible for parsing web socket messages and returning the resolved
 	// and unresolved prices.
-	PriceWebSocketDataHandler = wshandlers.WebSocketDataHandler[mmtypes.Ticker, *big.Int]
+	PriceWebSocketDataHandler = wshandlers.WebSocketDataHandler[ProviderTicker, *big.Int]
 
 	// PriceWebSocketQueryHandler is a type alias for the price web socket query handler.
 	// This is responsible for building the web socket query for the price provider and
 	// returning the resolved and unresolved prices.
-	PriceWebSocketQueryHandler = wshandlers.WebSocketQueryHandler[mmtypes.Ticker, *big.Int]
+	PriceWebSocketQueryHandler = wshandlers.WebSocketQueryHandler[ProviderTicker, *big.Int]
 
 	// PriceResponse is a type alias for the price response. A price response is
 	// composed of a map of resolved prices and a map of unresolved prices. Resolved
 	// prices are the prices that were successfully fetched from the API, while
 	// unresolved prices are the prices that were not successfully fetched from the API.
-	PriceResponse = providertypes.GetResponse[mmtypes.Ticker, *big.Int]
+	PriceResponse = providertypes.GetResponse[ProviderTicker, *big.Int]
```

Comment:
Is it safe to use an interface as a reference into a map (ifaces are reference types). Can't we just use the OffChainTicker string as the index into `ResolvedPrices` for each provider?

Reply:
I would check out the follow up PRs, but using only the offchain ticker as a primary key would not give us the amount of info we need to determine what we are parsing when we get a response.